python/cudf/cudf/_lib/cudf.pxd

```diff
@@ -382,4 +375,6 @@ cdef extern from "cudf/legacy/table.hpp" namespace "cudf" nogil:
 #        gdf_column const* const* end() const
 #        gdf_column const* get_column(size_type index) const except +
 
-cpdef gdf_dtype gdf_dtype_from_value(col, dtype=*) except? GDF_invalid
+cdef gdf_dtype gdf_dtype_from_dtype(dtype) except? GDF_invalid
+
+cdef char* py_to_c_str(object py_str)
```

Comment:
Handle exceptions?

Reply:
I wonder if it would be better to have this function just return `bytes`. That would allow the normal Python exception handling. Plus getting the underlying `char*` from a `bytes` object is fast. Not to mention Python optimizes a bunch of things with Python `bytes` that C doesn't do really well.

Reply:
I'll leave it as is, i.e., returning `char*` and add an `except?`. This function will be going away with `gdf_column` anyway